src/main/java/de/uol/pgdoener/civicsage/source/FileHashingService.java

```diff
@@ -0,0 +1,43 @@
+package de.uol.pgdoener.civicsage.source;
+
+import de.uol.pgdoener.civicsage.index.exception.ReadFileException;
+import de.uol.pgdoener.civicsage.source.exception.HashingException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HexFormat;
+
+@Slf4j
+@Service
+public class FileHashingService {
+
+    private static final String ALGORITHM = "SHA-256";
```

Comment:
I'm a bit puzzled, that there is no constant for referencing the hash algorithm. Just searched a little and all examples and documentation always use the hard-coded String.